crates/server/src/middleware/dev_auth.rs

```diff
@@ -0,0 +1,36 @@
+use std::sync::Arc;
+
+use axum::http::{Request, StatusCode};
+use axum::middleware::Next;
+use axum::response::Response;
+use axum::Extension;
+
+use crate::admin::service::AdminState;
+
+pub async fn dev_mode_auth(
+    state: Extension<Arc<AdminState>>,
+    request: Request<axum::body::Body>,
+    next: Next,
+) -> Result<Response, StatusCode> {
+    let public_key = &state.keypair.public();
+
+    let signature = request
+        .headers()
+        .get("X-Signature")
+        .and_then(|v| bs58::decode(v).into_vec().ok())
+        .ok_or(StatusCode::UNAUTHORIZED)?;
+
+    let timestamp = request
+        .headers()
+        .get("X-Timestamp")
+        .and_then(|v| Some(v.as_bytes()))
+        .ok_or(StatusCode::UNAUTHORIZED)?;
```

Comment:
I thought the use of timestamp here is to prevent replayability. There's no check for a timestamp threshold.

Maybe we permit a 5s threshold?